Grbl_Esp32/src/Error.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+  Error.cpp - Error names
+  Part of Grbl
+  Copyright (c) 2014-2016 Sungeun K. Jeon for Gnea Research LLC
+
+	2018 -	Bart Dring This file was modifed for use on the ESP32
+					CPU. Do not use this with Grbl for atMega328P
+        2020 - Mitch Bradley
+
+  Grbl is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  Grbl is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with Grbl.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "Error.h"
+
+std::map<Error, const char*> ErrorCodes = {
```

Comment:
My issue with this implementation is that you need two lists that you need to keep in sync. People are going to make mistakes with that, either by adding codes and forgetting to add an error code, or by adding something in the middle and messing with the numbers. 

I've thought about it some more, and think there's a better way to do this. This is better because the compiler will give an error, if you mess up with the above, for whatever reason. Minimum working example:

```
enum class Error
{
    OK = 0,
    X,
    Y, // Will automatically count 1, 2, ...
    NumberValues
};

const char* ErrorDescriptions[int(Error::NumberValues)] = {
    "OK",
    "X",
    "Y"
};

const char* get_error_description(/* Error value? */ int code)
{
    if (code >= 0 && code < int(Error::NumberValues))
    {
        return ErrorDescription[code];
    }
    else 
    {
        return nullptr;
    }
}
```

Reply:
Again, simplicity.  That map is not new, it was just moved into a better place.  Maybe we can figure out a better way to keep them in sync, but not in this go-around.